internal/fsutil/create_test.go

```diff
@@ -18,66 +18,68 @@ type createTest struct {
 	error   string
 }
 
-var createTests = []createTest{{
-	options: fsutil.CreateOptions{
-		Path:        "foo/bar",
-		Data:        bytes.NewBufferString("data1"),
-		Mode:        0444,
-		MakeParents: true,
-	},
-	result: map[string]string{
-		"/foo/":    "dir 0755",
-		"/foo/bar": "file 0444 5b41362b",
-	},
-}, {
-	options: fsutil.CreateOptions{
-		Path:        "foo/bar",
-		Link:        "../baz",
-		Mode:        fs.ModeSymlink,
-		MakeParents: true,
-	},
-	result: map[string]string{
-		"/foo/":    "dir 0755",
-		"/foo/bar": "symlink ../baz",
-	},
-}, {
-	options: fsutil.CreateOptions{
-		Path:        "foo/bar",
-		Mode:        fs.ModeDir | 0444,
-		MakeParents: true,
-	},
-	result: map[string]string{
-		"/foo/":     "dir 0755",
-		"/foo/bar/": "dir 0444",
-	},
-}, {
-	options: fsutil.CreateOptions{
-		Path: "tmp",
-		Mode: fs.ModeDir | fs.ModeSticky | 0775,
-	},
-	result: map[string]string{
-		"/tmp/": "dir 01775",
-	},
-}, {
-	options: fsutil.CreateOptions{
-		Path: "foo/bar",
-		Mode: fs.ModeDir | 0775,
-	},
-	error: `.*: no such file or directory`,
-}}
+func createTests() []createTest {
+	return []createTest{{
```

Comment:
Because the tests are creating readers inside them. If I use the tests as an array then the first tests empties the buffer and the second test that re-uses it fails. The other option is to change it so that we have raw bytes. I thought that this was more flexible but I dont have any strong stance.

Reply:
Ah okay, makes sense.

Reply:
The only benefit is that if we were using whole files here, not having a reader means storing the whole file in memory for the duration of the tests.